spec/thread_safety_spec.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+RSpec.describe "thread safety" do # rubocop:disable RSpec/DescribeClass
+  context "when two threads accessing unmemoized zero-args method" do
+    let(:thread_return_values) do
+      any_nils = ->(a) { a.any?(&:nil?) }
+
+      check_repeatedly(condition_proc: any_nils) do
+        instance = class_with_memo.new
+        threads = Array.new(2) { Thread.new { instance.return_first_thread_id } }
+        threads.map(&:value)
+      end
+    end
+
+    let(:class_with_memo) do
+      Class.new do
+        prepend MemoWise
+
+        def return_first_thread_id
```

Comment:
Nitpick: I wonder if a more "Rubyesque" name would just be `first_thread_id`.

Reply:
Done!